C7Engine/MapUnitExtensions.cs

```diff
@@ -54,8 +69,12 @@ public static bool fight(this MapUnit unit, MapUnit defender)
 				unit.hitPointsRemaining -= 1;
 		}
 
+		MapUnit loser = (defender.hitPointsRemaining <= 0) ? defender : unit,
+			winner = (defender == loser) ? unit : defender;
+
+		winner.RollToPromote(winner == unit, false);
```

Comment:
Shouldn't the first parameter be whether the unit was the attacker?  Looking at the parameter names for RollToPromote, I would expect it to be something like, `unit != defender`.

Reply:
It's correct as written, but I can see how it would be misleading. `unit` is always the attacker. I replaced that expression with `winner != defender` for clarity. When I was originally writing this code I briefly considered renaming `unit` to `attacker` but decided not to for consistency with the other extension methods, and laziness.

Reply:
Oh, yeah, I see it now, looking at it again.  Thanks for changing it though, it does make it more obvious.